### PR TITLE
Disallowed single quotes

### PR DESCRIPTION
**Pull request by Aditi-Singh16**
Closes issue #2094 

**To-do**
- [x] Disallow ' for wallets.
- [x] Disallow ' for devices 

**Demo**

https://user-images.githubusercontent.com/74586135/216130729-eb681c02-5ea7-4256-89eb-1e741a087814.mp4

### CODE REVIEW

File: src/cryptoadvance/specter/templates/wallet/new_wallet/new_wallet_keys.jinja [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 		<h3 class="mt-8">Information</h3>
 		<div class="floating-wrapper">
-			<input type="text" id="wallet_name" name="wallet_name" class="peer floating-input" value="{{ wallet_name }}" placeholder=" " required>
+			<input type="text" pattern="^[^']+" title="Do not use single quotes" id="wallet_name" name="wallet_name" class="peer floating-input" value="{{ wallet_name }}" placeholder=" " required>
```
[the remark this answers is not in the document]
Comment (Aditi-Singh16) [reply src/cryptoadvance/specter/templates/wallet/new_wallet/new_wallet_keys.jinja:12]: Yes I thought this too, but then it won't allow for the device name to be more than one character

**Comment by k9ert** [comment]
> KeyError: '20b0ac814c15b274fc34640151e8a6200601061b75455323af074cf72dfe5347'

the cypress-test failing is not related to this PR.

**Comment by moneymanolis** [comment]
@Aditi-Singh16 can you add a little Cypress test in `spec_devices.js`, always a good check when dealing with Regexes.

**Comment by Aditi-Singh16** [comment]
> a little Cypress test in `spec_devices.js`
Okay I will do it

**Comment by moneymanolis** [comment]
@Aditi-Singh16 the Cypress test should ideally be more of an integration test, so basically what a user will face when typing in a device name etc. So the idea would be:
Name a device with ', check for the right feedback, then name it correctly and check that the device was created successfully. You can delete the device then again.

**Comment by Aditi-Singh16** [comment]
> @Aditi-Singh16 the Cypress test should be more of an integration test, so basically what a user will face when typing in a device name etc. So the idea would be:
> 
> * Name a device with '
> * Check for the right error feedback
> * Then name it correctly and check that the device was created successfully
> * You can delete the device then again.

Updated

**Comment by moneymanolis** [comment]
Not really what I had in mind. Please have a close look at the existing tests.
Especially ` it('Electrum device')`
Looking at this, I am also wondering why this is not failing not with your changes?
`cy.get('#device_name').type("Electron's Electrum Device")`

There is an ' in it. Please investigate this.

**Comment by Aditi-Singh16** [comment]
> Not really what I had in mind. Please have a close look at the existing tests. Especially ` it('Electrum device')` Looking at this, I am also wondering why this line is not failing now with your changes? `cy.get('#device_name').type("Electron's Electrum Device")`
> 
> There is an ' in it. Please investigate this.

Okay, will look into this
I was trying to run tests however, I get this error
image

how do I fix this?

**Comment by Aditi-Singh16** [comment]
> Not really what I had in mind. Please have a close look at the existing tests. Especially ` it('Electrum device')` Looking at this, I am also wondering why this line is not failing now with your changes? `cy.get('#device_name').type("Electron's Electrum Device")`
> 
> There is an ' in it. Please investigate this.

Okay, I finally got it
The Electrum device was failing since the input field was different, now I have added a pattern check there too
so now it's showing the proper output
here's how it looks:

https://user-images.githubusercontent.com/74586135/216818744-7fc111b7-f6d5-4843-b012-e976ffe78486.mp4

**Comment by Aditi-Singh16** [comment]
I also changed the baseUrl in the cypress.json file to http://127.0.0.1:25441, but I didn't commit it

**Comment by k9ert** [comment]
Awesome! Thank you. Those are the really valuable PRs which even find the bugs in the other tests:
image

**Comment by Aditi-Singh16** [comment]
I think this PR is ready

If you think there are no more changes, could you please merge?

**Comment by k9ert** [comment]
I think the cypress-test is still failing. Check the tests, you have to adjust it as it's using an apostrophe.

**Comment by Aditi-Singh16** [comment]
> I think the cypress-test is still failing. Check the tests, you have to adjust it as it's using an apostrophe.

Oh yes, done
image

**Review by k9ert — APPROVED** [review]

**Comment by k9ert** [comment]
Thank you very much

**Pull request merged**